spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jooq/JooqExceptionTranslator.java

```diff
@@ -30,68 +30,17 @@
 import org.springframework.jdbc.support.SQLStateSQLExceptionTranslator;
 
 /**
- * Transforms {@link java.sql.SQLException} into a Spring-specific
+ * A {@link ExecuteListener} which can transforms or translate {@link Exception} into a Spring-specific
  * {@link DataAccessException}.
  *
- * @author Lukas Eder
- * @author Andreas Ahlenstorf
- * @author Phillip Webb
- * @author Stephane Nicoll
- * @since 1.5.10
+ * @author Dennis Melzer
+ * @since 3.2.1
  */
-public class JooqExceptionTranslator implements ExecuteListener {
-
-	// Based on the jOOQ-spring-example from https://github.com/jOOQ/jOOQ
-
-	private static final Log logger = LogFactory.getLog(JooqExceptionTranslator.class);
-
-	@Override
-	public void exception(ExecuteContext context) {
-		SQLExceptionTranslator translator = getTranslator(context);
-		// The exception() callback is not only triggered for SQL exceptions but also for
-		// "normal" exceptions. In those cases sqlException() returns null.
-		SQLException exception = context.sqlException();
-		while (exception != null) {
-			handle(context, translator, exception);
-			exception = exception.getNextException();
-		}
-	}
-
-	private SQLExceptionTranslator getTranslator(ExecuteContext context) {
-		SQLDialect dialect = context.configuration().dialect();
-		if (dialect != null && dialect.thirdParty() != null) {
-			String dbName = dialect.thirdParty().springDbName();
-			if (dbName != null) {
-				return new SQLErrorCodeSQLExceptionTranslator(dbName);
-			}
-		}
-		return new SQLStateSQLExceptionTranslator();
-	}
+public interface JooqExceptionTranslator extends ExecuteListener {
```

Comment:
This change breaks backwards compatability. Unfortunately `JooqExceptionTranslator` is public and therefore we need to either deprecate it and then replace it. Or we leave it in place as it is and let users subclass it if they want to change something.